human_general_array_analysis/1_process_data.py

```diff
@@ -0,0 +1,242 @@
+#
+# If i look up the genes that have NA, it says there is "no data" though I'm not sure what is causing this. So for now I will try to remove as many of the NAs as I can with filtering.
+
+# +
+# Matrix is sample x gene
+# Note, there are NaNs in the matrix. I'm not exactly sure what is causing this since the data
+# downloaded was based on what Crow et al used in their analysis and therefore should have been
+# filtered by platform.
+
+# Filter out samples that are all NaNs ***
+samples_to_drop = raw_compendium[
+    raw_compendium.isna().sum(axis=1) == raw_compendium.shape[1]
+].index
+processed_compendium = raw_compendium.drop(samples_to_drop)
+
+# All genes have at least 1 NaN so dropping all genes with NaN removes all the data
+# Instead we will move genes if they are NaN in _most_ samples (>90%)
+x = processed_compendium.isna().sum()
+processed_compendium = processed_compendium[x.index[x < 100]]
+
+# Remove remaining samples with NaNs
+processed_compendium = processed_compendium[processed_compendium.isna().sum(axis=1) < 1]
+
+assert processed_compendium.isna().sum().sum() == 0
+
+# +
+# Log transformed the data since the spread is very large
+processed_compendium_transform = np.log10(processed_compendium)
+
+# Replace -inf (from 0 input) and nans (from negative values) with 0
+# Not sure why there are negative values here.
+# Not sure what batch correction measures were used
+# TO DO: Check Gemma documentation again
+processed_compendium_transform = processed_compendium_transform.replace(
+    -np.inf, 0.0
+).replace(np.nan, 0.0)
+
+assert processed_compendium_transform.isna().sum().sum() == 0
+
+# +
+# Get only gene expression data for genes in Crow et. al.
+our_gene_ids_hgnc = list(processed_compendium_transform.columns)
+
+published_generic_genes = process.get_published_generic_genes(DE_prior_filename)
+shared_genes_hgnc = list(set(our_gene_ids_hgnc).intersection(published_generic_genes))
+
+# In Python, the order of elements in a list that is converted from a set
+# is non-deterministic, so it is sorted here to have reproducible result.
+shared_genes_hgnc.sort()
+
+# Pickle `shared_genes_hgnc` and save as `shared_genes_filename`
+if not os.path.exists(shared_genes_filename):
+    with open(shared_genes_filename, "wb") as pkl_fh:
+        pickle.dump(shared_genes_hgnc, pkl_fh)
+
+mapped_compendium = processed_compendium_transform[shared_genes_hgnc]
+
+assert mapped_compendium.isna().sum().sum() == 0
+
+print(mapped_compendium.shape)
+mapped_compendium.head(10)
+# -
+
+# Save
+mapped_compendium.to_csv(mapped_compendium_filename, sep="\t")
+
+# Normalize data
+process.normalize_compendium(
+    mapped_compendium_filename, normalized_compendium_filename, scaler_filename
+)
+
+# Check normalized data values
+normalized_compendium = pd.read_csv(
+    normalized_compendium_filename, sep="\t", index_col=0
+)
+
+normalized_compendium.head(10)
+
+sns.displot(normalized_compendium["AA06"])
```

Comment:
Is this distribution fairly typical of what you're seeing in other genes, or is there a reason you picked AA06 here?

Reply:
This was more me testing what the distribution of the data looked like after normalization - to make sure the values were between 0 and 1